src/controllers/userController.ts

```diff
@@ -62,4 +63,75 @@ const login = async (
   res.status(StatusCodes.OK).json({ user: { name: user.name }, token });
 };
 
-export { register, login };
+const getUserProfile = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> => {
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+
+    const { email } = req.body;
+
+    if (!email) {
+      return next(new BadRequestError("Please provide the email!"));
+    }
+
+    const userProfile = await User.findOne({ "email": email });
+
+    if (!userProfile) {
+      return next(new NotFoundError("The user with such email was not found"));
+    }
+
+    res.status(StatusCodes.OK).json({ name: userProfile.name, email: userProfile.email, id: userProfile._id });
+  } catch (error) {
+    return next(error);
+  }
+
+};
+
+const updateUserProfile = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> => {
+  //password could be added later if we'll decide to provide this option to the user
+  const { body: { name, email }, params: { id: userIdToBeUpdated } } = req;
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  //depending on the Front we can change the approach and compare by e.g. email
+  if (user.userId.toString() !== userIdToBeUpdated) {
```

Comment:
same comment as above, if the authenticated profile is been updated, then use the identifier from the authenticated user when updating the db

Reply:
Fixed. Now the only option to update a profile is to update your own profile, no way to update another profile